r/R/arrow-package.R

```diff
@@ -196,6 +196,20 @@ configure_tzdb <- function() {
         )
       )
     }
+
+    # On MacOS only, Check if we are running in under emulation, and warn this will not work
+    rosetta <- identical(sysname, "darwin") && identical(system("sysctl -n sysctl.proc_translated", intern = TRUE), "1")
+    if (rosetta) {
+      packageStartupMessage(
+        paste(
+          "It appears that you are running R and Arrow in emulation",
+          "(typically, this is running an x86 build of R on an ARM or",
+          "M processor based mac). This configuration is not supported",
+          "by arrow and we highly recommend you install a native (arm64)",
+          "build of R and use arrow with that."
```

Comment:
This is more verbose than many of our messages (and I'm happy for some streamlining), but my reasoning for being _so_ explicit is that if someone finds themselves in this situation, they might not even realized what is going on with architectures, versions, emulation etc. So I wanted to give people more details to be able to get to a good state (native R + native arrow)

Reply:
I might explicitly link to https://cran.r-project.org/bin/macosx/ and say install the arm64 version. And I (personally) would be a bit more direct: you're running R built for Intel on a non-Intel mac, and arrow won't work in that context; to fix, install the arm64 version of R from CRAN and then reinstall R.

Reply:
Thanks for the "Intel on a non-Intel mac" that's probably the clearest way to describe it to folks who are most likely to run into this.